### PR TITLE
Fix parameters and docs in metrics

### LINKED ISSUE #2377: Incorrect docs in metrics
## 📚 Documentation
Here
https://github.com/PyTorchLightning/pytorch-lightning/blob/a5f45787eabddfec4559983f8e6ba1c8317f62f1/pytorch_lightning/metrics/functional/classification.py#L137
the argument is `num_classes` but in docs it is `class_index`
https://github.com/PyTorchLightning/pytorch-lightning/blob/a5f45787eabddfec4559983f8e6ba1c8317f62f1/pytorch_lightning/metrics/functional/classification.py#L148

Also, it returns 5 values but only 4 are mentioned.
https://github.com/PyTorchLightning/pytorch-lightning/blob/a5f45787eabddfec4559983f8e6ba1c8317f62f1/pytorch_lightning/metrics/functional/classification.py#L152-L153

**Also, some more typos and docs needed to be fixed in the metrics module.**

### PR DESCRIPTION
**Pull request by rohitgr7**
## What does this PR do?

Fixes #2377

# Before submitting

- [x] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together? Otherwise, we ask you to create a separate PR for every change.
- [x] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests? 
- [x] Did you verify new and existing tests pass locally with your changes?
- [ ] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)?

## PR review    
Anyone in the community is free to review the PR once the tests have passed.     
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: pytorch_lightning/metrics/functional/classification.py [hunk 1]
```diff
@@ -105,10 +107,10 @@ def stat_scores(
         target: target tensor
         class_index: class to calculate over
         argmax_dim: if pred is a tensor of probabilities, this indicates the
-            axis the argmax transformation will be applied over
+            axis the argmax transformation will be applied over (default: 1)
 
     Return:
-        True Positive, False Positive, True Negative, False Negative
+        True Positive, False Positive, True Negative, False Negative, SUP
```
Comment (rohitgr7) [inline pytorch_lightning/metrics/functional/classification.py:113]: I know what this is but not sure what to write here `SUP`.

Comment (SkafteNicki) [reply pytorch_lightning/metrics/functional/classification.py:113]:
```suggestion
        True Positive, False Positive, True Negative, False Negative, Support
```

File: pytorch_lightning/metrics/functional/classification.py [hunk 2]
```diff
@@ -145,12 +147,12 @@ def stat_scores_multiple_classes(
...
         target: target tensor
-        class_index: class to calculate over
+        num_classes: number of classes if known (default: None)
         argmax_dim: if pred is a tensor of probabilities, this indicates the
-            axis the argmax transformation will be applied over
+            axis the argmax transformation will be applied over (default: 1)
 
     Return:
-        True Positive, False Positive, True Negative, False Negative
+        True Positive, False Positive, True Negative, False Negative, SUPS
```
Comment (SkafteNicki) [inline pytorch_lightning/metrics/functional/classification.py:155]:
```suggestion
        True Positive, False Positive, True Negative, False Negative, Support
```

File: pytorch_lightning/metrics/functional/classification.py [hunk 3]
```diff
@@ -856,19 +868,24 @@ def dice_score(
...
 ) -> torch.Tensor:
     """
+    Compute dice score from prediction scores
+
     Args:
         pred: estimated probabilities
         target: ground-truth labels
-        bg: whether to also compute dice for the background
+        bg: whether to also compute dice for the background (default: False)
         nan_score: score to return, if a NaN occurs during computation (denom zero)
```
Comment (Borda) [inline pytorch_lightning/metrics/functional/classification.py:877]:
```suggestion
        nan_score: score to return, if a NaN occurs during computation (default zero)
```

**Comment by Borda** [comment]
just curious, shall be these default taken from annotation/typing?

**Comment by awaelchli** [comment]
@Borda sphinx does that anyway and writes it in the signature. So if we write the defaults in the docstring also, we risk it getting outdated if someone changes the default in the signature but forgets to update the docstring or vice versa. On the other hand, it is a preference thing, so I don't know. I think it should be decided which way we go and then consequently be done everywhere. Since docs are still changing a lot, I wouldn't worry too much about it.

**Comment by Borda** [comment]
@awaelchli exactly that was my point that we shall not duplicate it and rather use the automatic if possible... 
so you recommend dropping this PR, eventually remove all defaults from docstrings?

**Comment by awaelchli** [comment]
Yes, I think more arguments speak for the automatic method. Also, annotations like "default: None" are not really helpful since they can lead to special behaviours that need to be explained, otherwhise it's not very informative. My view is that simplicity and consistency leads to more clarity.

**Pull request merged**